src/python/pants/backend/scala/dependency_inference/ScalaParser.scala

```diff
@@ -8,14 +8,20 @@ import scala.meta._
 import scala.meta.transversers.Traverser
 
 import scala.collection.mutable.ArrayBuffer
+import scala.collection.mutable.HashMap
+
+case class AnImport(name: String, isWildcard: Boolean)
 
 case class Analysis(
-  providedNames: Vector[String]
+  providedNames: Vector[String],
+  importsByNamespace: HashMap[String, ArrayBuffer[AnImport]],
```

Comment:
bike shed: I called this "by namespace" but maybe "by scope" would be better?

Reply:
Probably by scope, yea: https://www.scala-lang.org/files/archive/spec/2.13/02-identifiers-names-and-scopes.html